Firmware/ultralcd.cpp

```diff
@@ -6675,45 +6675,51 @@ static void lcd_tune_menu()
 	MENU_END();
 }
 
-static void mbl_magnets_elimination_set() {
+static void mbl_magnets_elimination_toggle() {
 	bool magnet_elimination = (eeprom_read_byte((uint8_t*)EEPROM_MBL_MAGNET_ELIMINATION) > 0);
 	magnet_elimination = !magnet_elimination;
 	eeprom_update_byte((uint8_t*)EEPROM_MBL_MAGNET_ELIMINATION, (uint8_t)magnet_elimination);
 }
 
-static void mbl_mesh_set() {
+static void mbl_mesh_toggle() {
 	uint8_t mesh_nr = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
 	if(mesh_nr == 3) mesh_nr = 7;
 	else mesh_nr = 3;
 	eeprom_update_byte((uint8_t*)EEPROM_MBL_POINTS_NR, mesh_nr);
 }
 
+static void mbl_probe_nr_set() {
+	mbl_z_probe_nr = eeprom_read_byte((uint8_t*)EEPROM_MBL_PROBE_NR);
+	switch (mbl_z_probe_nr) {
+		case 1: mbl_z_probe_nr = 3; break;
+		case 3: mbl_z_probe_nr = 5; break;
+		case 5: mbl_z_probe_nr = 1; break;
+		default: mbl_z_probe_nr = 3; break;
+	}
+	eeprom_update_byte((uint8_t*)EEPROM_MBL_PROBE_NR, mbl_z_probe_nr);
+}
+
 static void lcd_mesh_bed_leveling_settings()
 {
 	
 	bool magnet_elimination = (eeprom_read_byte((uint8_t*)EEPROM_MBL_MAGNET_ELIMINATION) > 0);
 	uint8_t points_nr = eeprom_read_byte((uint8_t*)EEPROM_MBL_POINTS_NR);
 
 	MENU_BEGIN();
-	// leaving menu - this condition must be immediately before MENU_ITEM_BACK_P
-	if (((menu_item == menu_line) && menu_clicked && (lcd_encoder == menu_item)) || menu_leaving)
-	{
-		eeprom_update_byte((uint8_t*)EEPROM_MBL_PROBE_NR, (uint8_t)mbl_z_probe_nr);
-	}
 	MENU_ITEM_BACK_P(_T(MSG_SETTINGS)); 
-	if(points_nr == 3) MENU_ITEM_FUNCTION_P(_i("Mesh         [3x3]"), mbl_mesh_set);
-	else			   MENU_ITEM_FUNCTION_P(_i("Mesh         [7x7]"), mbl_mesh_set);
-	MENU_ITEM_EDIT_int3_P(_i("Z-probe nr."), &mbl_z_probe_nr, 1, 5);
+	if(points_nr == 3) MENU_ITEM_FUNCTION_P(_i("Mesh         [3x3]"), mbl_mesh_toggle); ////MSG_MESH_3x3 c=18 r=0
```

Comment:
Hi @PavelSindler ... i started to cleanup some messages in the firmware prusa3d#1644
Is the 'messages.c/h' still used or not? If so wouldn't it make sense to add 'MSG_MESH_3x3' to these and change the link here?